src/mpid/ch4/netmod/ofi/ofi_impl.h

```diff
@@ -194,33 +194,6 @@ int MPIDI_OFI_progress(int vci, int blocking);
         MPIR_Request_add_ref((req));                                \
     } while (0)
 
-MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_need_request_creation(const MPIR_Request * req)
-{
-    if (MPIDI_CH4_MT_MODEL == MPIDI_CH4_MT_DIRECT) {
-        return 1;       /* Always allocated by netmod */
-    } else if (MPIDI_CH4_MT_MODEL == MPIDI_CH4_MT_HANDOFF) {
-        return (req == NULL);
-    } else {
-        /* Invalid MT model */
-        MPIR_Assert(0);
-        return -1;
-    }
-}
-
-#define MPIDI_OFI_REQUEST_CREATE_CONDITIONAL(req, kind, vni) \
```

Comment:
Can this macro instead contain the code for creating requests that is now repeated in multiple places? It appears to me there's still some use in wrapping that code up cleanly vs. having ifdefs and branches in all the calling functions.

Reply:
The workq/handoff code is not final yet. Right now I am just pushing it aside, and I would rather work with the `ifdef/branch` explicitly so I can easily be conscious about it rather than hiding it behind the macro. But in general, yes, I agree that this may be refactored (back) later.